pkg/operation/botanist/health_check.go

```diff
@@ -652,14 +668,29 @@ func (b *Botanist) checkControlPlane(
 // checkSystemComponents checks whether the system components of a Shoot are running.
 func (b *Botanist) checkSystemComponents(
 	ctx context.Context,
+	gardenerVersion string,
 	checker *HealthChecker,
 	condition gardencorev1beta1.Condition,
 	shootDeploymentLister kutil.DeploymentLister,
 	shootDaemonSetLister kutil.DaemonSetLister,
 	extensionConditions []ExtensionCondition,
 ) (*gardencorev1beta1.Condition, error) {
 
-	if exitCondition, err := checker.CheckSystemComponents(metav1.NamespaceSystem, condition, shootDeploymentLister, shootDaemonSetLister); err != nil || exitCondition != nil {
+	gardenerVersionGeq140, err := versionutils.CheckVersionMeetsConstraint(gardenerVersion, ">= v1.5.0")
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		konnectivityHealthCheck = b.Shoot.KonnectivityTunnelEnabled && gardenerVersionGeq140
```

Comment:
Hm, another thing came to my mind here: Assume somebody enables the feature gate and Gardener reconciles the cluster --> konnectivity gets rolled out. Now he disables the feature gate again. Then this check will make Gardener check the VPN-shoot instead of the currently deployed konnectivity. Only after <= 24 hours (in case reconciliation only happens in maintenance time window) it will resolve, but in the meantime it will report a false negative health check.

As this feature is configurable (which e.g. wasn't the case for etcd-druid) this check here is not good enough. Sorry for not thinking about it earlier. Probably it's better to store the information which VPN solution is deployed somewhere in the seed. For example, one solution could be to let the shoot reconciliation flow annotate the `shoot-core` resource and read it here to decide what you have to check.